_monitoring-plugins/ad/result-mapping.md

```diff
@@ -0,0 +1,319 @@
+      "data": 0.03384479053991436
+    },
+    {
+      "feature_id": "processing_bytes_min",
+      "data": 0.058812549572819096
+    },
+    {
+      "feature_id": "processing_bytes_sum",
+      "data": 0.10154576265526988
+    },
+    {
+      "feature_id": "processing_time_max",
+      "data": 0.7695105170276828
+    }
+  ],
+  "expected_values": [
+    {
+      "likelihood": 1,
+      "value_list": [
+        {
+          "feature_id": "processing_bytes_max",
+          "data": 2291
+        },
+        {
+          "feature_id": "processing_bytes_avg",
+          "data": 1677.3333333333333
+        },
+        {
+          "feature_id": "processing_bytes_min",
+          "data": 1054
+        },
+        {
+          "feature_id": "processing_bytes_sum",
+          "data": 6062
+        },
+        {
+          "feature_id": "processing_time_max",
+          "data": 23379
+        }
+      ]
+    }
+  ],
+  "threshold": 1.0993584705913992,
+  "execution_end_time": 1635898427895,
+  "execution_start_time": 1635898427803
+}
+```
+
+You can see the following additional fields:
+
+Field | Description
+:--- | :---
+`relevant_attribution` | Represents the contribution of each input variable. The sum of the attributions is normalized to 1.
+`expected_values` | The expected value for each feature.
+
+At times, the detector might detect an anomaly late.
+Let's say the detector sees a random mix of the triples {1, 2, 3} and {2, 4, 5} that correspond to `slow weeks` and `busy weeks`, respectively. For example 1, 2, 3, 1, 2, 3, 2, 4, 5, 1, 2, 3, 2, 4, 5, ... and so on.
+If the detector comes across a pattern {2, 2, X} and it's yet to see X, the detector infers that the pattern is anomalous, but it can't determine at this point which of the 2's is the cause. If X = 3, then the detector knows it's the first 2 in that unfinished triple, and if X = 5, then it's the second 2. If it's the first 2, then the detector detects the anomaly late.
+
+If a detector detects an anomaly late, the result has the following additional fields:
+
+Field | Description
+:--- | :---
+`past_values` | The actual input that triggered an anomaly. If `past_values` is null, the attributions or expected values are from the current input. If `past_values` is not null, the attributions or expected values are from a past input (for example, the previous two steps of the data [1,2,3]).
+`approx_anomaly_start_time` | The approximate time of the actual input that triggers an anomaly. This field helps you understand when a detector flags an anomaly. If the data is not continuous, the actual time that the detector detects the anomaly can be earlier. Single-stream detectors query previous anomaly results of a few data points while high-cardinality detectors don't do this because querying previous results is an expensive operation when performed on a lot of entities. So, the accuracy of this field is less for high-cardinality detectors as compared to single-stream detectors.
```

Comment:
There is no difference between single stream and hcad.  Both of them does not query because of the expensive queries.

Reply:
Updated. Please review.